redhat-marketplace/redhat-marketplace-operator
# Data Service upload target

**What's Changed**
- Adds a new meterreport upload target on the reporter tool for the Data Service and integrates the data service's grpc client 
- Allows for multiple upload targets to be set as a comma seperated list on the meterreport's `extraJobArgs` field:
```
...
  extraJobArgs: ["--uploadTargets=data-service,redhat-insights"]
 ```
 - Adds logic to `ReporterJob()` in factory: if the dataservice is enabled on the Meterbase cr, factory builds the meterreport job with volumes for the `serving-certs-ca-bundle` cm and the service account token for the data service service.
 
 **How to Test**
- Edit the meterbase cr to endable the Data Service `dataServiceEnabled: true`
- Wait for the dqlite cluster to become available
- Apply a meterdefinition and a metterreport with correct extra args from above
- Check the report job's logs - you should see: 
Correct upload target being set and a response from the data service api with the file that was just uploaded
```
upload target	{"target set to": "data-service"}
...
airgap upload response  {"response": "file_id:{name:\"upload-bb3bb078-11a5-4c86-b940-817b5061b86f.tar.gz\"} size:387"}

```
